### ISSUE TITLE
Directory permissions issue on Windows

### ISSUE THREAD

**Issue by Maowse** [balderdashy/skipper-s3#35]
We are able to upload files to our s3 bucket from a machine running OSX, but running the same code on a Windows 7 machine results in the following error:

![image](https://cloud.githubusercontent.com/assets/9093806/10249541/e0ff71e2-68f3-11e5-9c8e-b23369ecffa2.png)

For some reason, it appears that the Windows machine has issues accessing the intermediate directory that skipper-s3 makes when a dirname is specified. If we remove the dirname property, the file is uploaded with no issues.

We use the following code to upload:

```
req.file('receipt').upload({
               adapter: require('skipper-s3'),
               key: 'XXXXXXXXX',
               secret: 'XXXXXXXX',
               bucket: 'someBucket',
               region: "us-east-1",
               dirname: 'someDirectory'
           }
```

**Comment by sailsbot** [comment]
Thanks for posting, @Maowse.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by mattcodez** [comment]
@Maowse  Did you happen to find a solution for this? My current thought is that I'll have to compute my own filename hash and pass a function to `saveAs` so that I can specify an S3 directory.

**Comment by Maowse** [comment]
No, unfortunately. I believe we just ended up using a base bucket for now.
